Automattic/wp-calypso
# Social Previews | Add Threads preview

## Proposed Changes

* Add Threads preview to `social-previews` package
* Add Threads preview to post share preview

## Why are these changes being made?


* We are adding Threads support to Jetpack Social and we want to display post preview for Threads.

## Testing Instructions

* Sandbox D152810-code and follow the instructions given there
* Point `public-api.wordpress.com` to your sandbox
* Goto `/posts/:site`
* For any post that you want to check preview for, click on options (3 dots)
* Click on Share and then click on Preview
* Confirm that you see the Threads tab
* Select Threads tab 
* Confirm that the preview works as expected in different scenarios
* Please also test this in combination with https://github.com/Automattic/jetpack/pull/38003

## Pre-merge Checklist




When there is NO Threads connection but an image
<img width="707" alt="Screenshot 2024-06-24 at 3 36 04 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/70b234b9-0d89-402d-92bd-34872f08dcd9">

When there is NO Threads connection and No image
<img width="701" alt="Screenshot 2024-06-24 at 3 36 55 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/06c5958d-e38c-4c01-8a0d-00559b787fbf">

When there IS a Threads connection but and also an image
<img width="781" alt="Screenshot 2024-06-24 at 3 39 10 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/20553f35-2a03-4e7d-a2a2-f0555c626da2">

When there IS a Threads connection but NO image
<img width="701" alt="Screenshot 2024-06-24 at 3 40 30 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/e789e85b-63aa-4761-bb44-ed1eea4de82f">

When custom media is added to the post with "Share as social post" enabled
<img width="728" alt="Screenshot 2024-06-24 at 3 46 02 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/71904508-c167-46a0-8b66-c791486472ad">
